### ISSUE TITLE
Create Core Simulation Systems

### ISSUE THREAD

**Issue by Gerrudo** [Gerrudo/reticulating-splines#8]
Core Simulation systems need to be added to the game, these include:

- Economics
- Power 
- Water
- Employment

To start, these systems will be a availability vs demand, so the availability must either match or exceed the demand for the city to function.

The calculation for this can be done each second or on a turn based system.

**Comment by Gerrudo** [comment]
Putting on hold as this now depends on #10

**Comment by Gerrudo** [comment]
> Putting on hold as this now depends on [#10](https://github.com/Gerrudo/reticulating-splines/issues/10)

No longer on hold, simulation now works in part.

Money is now being deducted per second depending on the current buildings placed.

Items that still need work:

- Unsure if the total money is being deducted if a building is being placed
- Apartment buildings are not affecting employment figures
- Population also does not seem to be increasing
- **Should we look at introducing power/water usage at this stage?**

**Comment by Gerrudo** [comment]
- ~Unsure if the total money is being deducted if a building is being placed~
- Apartment buildings are not affecting employment figures
- Population also does not seem to be increasing
- ~Buildings can also be placed if there is not enough money left (SimulationManager needs to prevent this)~
- Should we look at introducing power/water usage at this stage?

As of now we've converted over to bricks, to simply the game loop, will detail this in another ticket.
